documentation/src/main/asciidoc/topics/con_jvm_heap_memory_allocation.adoc

```diff
@@ -1,26 +1,100 @@
 [id='jvm-heap-memory-allocation_{context}']
 = JVM heap space allocation
 
-When allocating JVM heap space for {brandname} deployments, you can use the following guidelines:
+Determine the amount of JVM memory that is required for your {brandname} deployment, so that you have sufficient data storage capacity to meet your needs.
+
+[IMPORTANT]
+====
+Allocating too much of JVM heap space for data storage can lead to more frequent garbage collection (GC) and GC pauses.
+These operations can lead to {brandname} cluster instability and network partitions.
+====
+
+The following table outlines two examples of JVM heap space allocation scenarios:
 
 [%autowidth,cols="1,1",stripes=even,%noheader]
 |===
 |Cache operations only (read, write, delete)
-|Allocate **50%** of JVM heap space for data storage
-
+|Allocate **50%** of JVM heap space for data storage.
 |Cache operations + data processing such as queries and cache event listeners
-|Allocate **33%** of JVM heap space for data storage
+|Allocate **33%** of JVM heap space for data storage.
 |===
 
-[IMPORTANT]
-====
-Allocating too much of JVM heap space for data storage can lead to more frequent garbage collection (GC) as well as GC pauses that lead to {brandname} cluster instability and network partitions.
-====
+The scenarios outlined in the previous table considered the following factors when allocating JVM heap space:
+
+* Clustered cache mode
+* Number of segments
+** For example, a low number of segments might affect how a server distributes data across nodes.
+* Read or write operations
+* Rebalancing requirements
+** For example, a high number of threads might quickly run in parallel during a state transfer, but each thread operation might consume more memory. 
+* Scaling clusters
+* Synchronous or asynchronous replication
+
+Consider the previous list of items when determining the amount of memory and CPU requirements for your {brandname} deployment. 
+Most notable {brandname} operations that require higher CPU limits include rebalancing nodes after pod restarts, running indexing queries on data, and performing GC operations. 
+ 
+The following example calculation helps determine the amount JVM memory required for an {brandname} deployment:
+
+[source,options="nowrap",subs=”+attributes.+quotes”]
+----
+JVM memory = JVM code + JVM stack size + JVM heap space
+----
+
+Note the following elements from the previous calculation:
+* A JVM stack, also called a thread stack, is an area of JVM memory that stores JVM threads. 
```

Comment:
to be preceise not threads but the data used by threads (maybe see JVM documentation for more...)

Reply:
Thanks. Updated.